javascripts/services/ContactsService.js

```diff
@@ -0,0 +1,41 @@
+"use strict";
+
+app.service("ContactsService", function($http, $q, FIREBASE_CONFIG){
+	const getAllContacts = (userUid) => {
+		let contacts = [];
+		return $q((resolve, reject) => {
+			$http.get(`${FIREBASE_CONFIG.databaseURL}/contacts.json?orderBy="uid"&equalTo="${userUid}"`).then((results) => {
+				let fbContacts = results.data;
+				console.log("results", results);
+				Object.keys(fbContacts).forEach((key) => {
+					fbContacts[key].id = key;
+					contacts.push(fbContacts[key]);
+					resolve(contacts);
```

Comment:
make sure to resolve your contacts after you completed your forEach loop through the Object.keys of your fbContacts